src/app/static/view-question/view-question.component.ts

```diff
@@ -17,25 +15,48 @@ export class ViewQuestionComponent implements OnInit {
    * on the site.
    */
   selectedQuestion: any;
-  responses: Response[];
+  highlightedResponse: any;
+  questionResponses: any;
 
   constructor(
-    private responseService: ResponseService,
     private questionService: QuestionService,
   ) {}
 
   ngOnInit() {
+    const selectedQuestionId = this.questionService.getQuestionId();
+
     this.questionService.getQuestions().subscribe(questions => {
       for (const q in questions) {
-        if (this.questionService.getQuestionId() === questions[q].id) {
+        if (selectedQuestionId === questions[q].id) {
           this.selectedQuestion = questions[q];
         }
       }
```

Comment:
why not just get the question by its id?  You have an endpoint for that

Reply:
Refactored to get question by the ID of the selected question